src/pages/EditRequestDistancePage.js

```diff
@@ -0,0 +1,111 @@
+import React, {useEffect, useRef} from 'react';
+import lodashGet from 'lodash/get';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import CONST from '../CONST';
+import ScreenWrapper from '../components/ScreenWrapper';
+import HeaderWithBackButton from '../components/HeaderWithBackButton';
+import Navigation from '../libs/Navigation/Navigation';
+import useLocalize from '../hooks/useLocalize';
+import DistanceRequest from '../components/DistanceRequest';
+import reportPropTypes from './reportPropTypes';
+import * as IOU from '../libs/actions/IOU';
+import transactionPropTypes from '../components/transactionPropTypes';
+import * as TransactionEdit from '../libs/actions/TransactionEdit';
+
+const propTypes = {
+    /** The transactionID we're currently editing */
+    transactionID: PropTypes.string.isRequired,
+
+    /** The report to with which the distance request is associated */
+    report: reportPropTypes.isRequired,
+
+    /** Passed from the navigator */
+    route: PropTypes.shape({
+        /** Parameters the route gets */
+        params: PropTypes.shape({
+            /** Type of IOU */
+            iouType: PropTypes.oneOf(_.values(CONST.IOU.MONEY_REQUEST_TYPE)),
+            /** Id of the report on which the distance request is being created */
+            reportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    /* Onyx props */
+    /** The original transaction that is being edited */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function EditRequestDistancePage({report, route, transaction}) {
+    const {translate} = useLocalize();
+    const transactionWasSaved = useRef(false);
+    const hasWaypointError = useRef(false);
+
+    useEffect(() => {
+        hasWaypointError.current = Boolean(lodashGet(transaction, 'errorFields.route') || lodashGet(transaction, 'errorFields.waypoints'));
+    }, [transaction]);
+
+    useEffect(() => {
+        // When the component is mounted, make a backup copy of the original transaction that is stored in onyx
+        TransactionEdit.createBackupTransaction(transaction);
+
+        return () => {
+            // When the component is unmounted
+            // If the transaction was saved without errors
+            // Then remove the backup transaction because it is no longer needed
+            if (transactionWasSaved.current && !hasWaypointError.current) {
+                TransactionEdit.removeBackupTransaction(transaction.transactionID);
+                return;
+            }
+
+            // If the transaction had errors, or wasn't saved, then the original transaction
+            // needs to be restored or else errors and edited fields will remain in the UI
+            TransactionEdit.restoreOriginalTransactionFromBackup(transaction.transactionID);
```

Comment:
I don't really understand why we're removing the backup or restoring the transaction here.

Can you give an example of the flow where this is used?

If there was an error editing a distance request I would image that the error messages should stick around until they are explicity dismissed, at which point we would restore the original transaction.

Reply:
>You cancel out of the modal: original is restored, backup is deleted

Sounds good
>You save the changes with no errors: backup is deleted

That makes sense, but I think that should happen as part of the success data. We're talking about backend errors right?

> You save the changes, there are errors: do nothing

So if you edited a distance request with invalid waypoints while offline, go online, and errors are returned, then we should leave the invalid data and error messages in Onyx until you edit them and try again. The only way to clear the errors would be to complete the edit with valid data. Or alternatively dismissing the errors could reset the data. I'm not sure on the UX...

So I guess we should only restore the backup when closing the modal, and delete the backup if we hit save, errors or not.